Assets/Scripts/Teleport/HandleHandTeleport.cs

```diff
@@ -0,0 +1,51 @@
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit.Interactors;
+using UnityEngine.XR.Interaction.Toolkit.Locomotion.Teleportation;
+
+namespace RogueApeStudios.SecretsOfIgnacios.Teleport
+{
+    public class HandleHandTeleport : MonoBehaviour
+    {
+        [SerializeField] private XRRayInteractor _teleportInteractor;
+
+        [SerializeField] private TeleportationProvider _teleportationProvider; 
+        // This is the "Teleport" gameobject under the "Locomotion" gameobject in the XROrigin
+
+        private bool _isTeleportActive = false;
+
+        private void Start()
+        {
+            _teleportInteractor.gameObject.SetActive(false);
+        }
+
+        public void ActivateTeleport()
+        {
+            if (!_isTeleportActive)
+            {
+                _teleportInteractor.gameObject.SetActive(true);
```

Comment:
Try to use references, I know it's the same object, but unity uses less operations with a direct reference.

```
[SerializeField] private XRRayInteractor _teleportInteractor;
[SerializeField] private GameObject _interactorObject;

_interactorObject.SetActive(true);
```